master/internal/resourcemanagers/agent/agent_state.go

```diff
@@ -255,6 +255,35 @@ func (a *AgentState) agentStarted(ctx *actor.Context, agentStarted *aproto.Agent
 	}
 }
 
+func (a *AgentState) checkAgentStartedDevicesMatch(
+	ctx *actor.Context, agentStarted *aproto.AgentStarted,
+) error {
+	ourDevices := map[device.ID]device.Device{}
+	for did, slot := range a.slotStates {
+		ourDevices[did] = slot.device
+	}
+
+	theirDevices := map[device.ID]device.Device{}
+	for _, d := range agentStarted.Devices {
+		theirDevices[d.ID] = d
+	}
+
+	if len(ourDevices) != len(theirDevices) {
```

Comment:
this logic (maps.Equal with better messaging) is in enough places it may make sense to pull out. the testing assert package has it but wouldn't want use that here I don't think..

Reply:
that's the only place we use `maps.Equal` based on search? I imagine we may be comparing maps by hand in some code which would already benefit just from a move to `maps.Equal`. same for `maps.Keys`.